src/components/tl/nccl/tl_nccl.h

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2021.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCC_TL_NCCL_H_
+#define UCC_TL_NCCL_H_
+
+#include "components/tl/ucc_tl.h"
+#include "components/tl/ucc_tl_log.h"
+#include "utils/ucc_mpool.h"
+
+#include <nccl.h>
+#include <cuda.h>
+
+typedef struct ucc_tl_nccl_iface {
+    ucc_tl_iface_t super;
+} ucc_tl_nccl_iface_t;
+
+extern ucc_tl_nccl_iface_t ucc_tl_nccl;
+
+typedef struct ucc_tl_nccl_lib_config {
+    ucc_tl_lib_config_t super;
+} ucc_tl_nccl_lib_config_t;
+
+typedef struct ucc_tl_nccl_context_config {
+    ucc_tl_context_config_t super;
+} ucc_tl_nccl_context_config_t;
+
+typedef struct ucc_tl_nccl_lib {
+    ucc_tl_lib_t super;
+} ucc_tl_nccl_lib_t;
+UCC_CLASS_DECLARE(ucc_tl_nccl_lib_t, const ucc_base_lib_params_t *,
+                  const ucc_base_config_t *);
+
+typedef struct ucc_tl_nccl_context {
+    ucc_tl_context_t             super;
+    ucc_tl_nccl_context_config_t cfg;
+    ucc_mpool_t                  req_mp;
+} ucc_tl_nccl_context_t;
+UCC_CLASS_DECLARE(ucc_tl_nccl_context_t, const ucc_base_context_params_t *,
+                  const ucc_base_config_t *);
+
+typedef struct ucc_tl_nccl_team {
+    ucc_tl_team_t        super;
+    ncclUniqueId        *unique_id;
+    void                *oob_req;
+    ucc_team_oob_coll_t  oob;
```

Comment:
i guess oob might be needed in multiple places as well. Does it makes sense to save it on ucc_base_team_t ?
Alternatively, we already have ucc_team_params_t (which has oob) on core ucc_team_t. Maybe we should have a pointer to core team on the ucc_base_team. Then we store oob there on params and all tls/cls have access to it.